### DISCUSSION TITLE
提出API関係

### DISCUSSION THREAD

**Discussion by namikis** [jphacks/F_2109#27] [Q&A]
@kenkenpa2126 @nishidayoshikatsu 
提出APIの件で3点お伺いしたいです。

１．miroの方ではfileオブジェクトのcodeプロパティがstring[]になってるんですが、これって配列で合ってますか？
２．「ディレクトリを配置して静的に公開（nodeのルーティングに含めない）してルートディレクトリを返す」という認識で大丈夫でしょうか？
３．↓のオブジェクトでテストしているのですが、構造的にはこれで合ってますか？

       const index4 = {
            fileName: "index4.html",
            code: "<h1>index4: testtesttest.</h1>"
        };

        const dir3 = {
            dirName: "dir3",
            file: [index4]
        };

        const index3 = {
            fileName: "index3.html",
            code: "<h1>index3: testtesttest.</h1>"
        };
        const dir2 = {
            dirName: "dir2",
            file: [index3]
        };
        const index2 = {
            fileName: "index2.html",
            code: "<h1>index2: testtesttest.</h1>"
        };
        const dir1 = {
            dirName: "dir1",
            dir: [dir2, dir3],
            file: [index2]
        };
        const main = {
            fileName: "main.html",
            code: "<h1>main: testtesttest.</h1>"
        };
        const index = {
            fileName: "index.html",
            code: "<h1>index: testtesttest.</h1>"
        };
        
        const source_code = {
            dirname: "/",
            file: [index, main],
            dir: [dir1]
        };

**Comment by kenkenpa2126** [comment]
1. かつてはcodeの中に複数のコードを配列で格納しようとしてたんですが、今は一つのfileオブジェクトにつき一つになっていて、配列ではなく単なる文字列です。stringの配列ということで文字列的なお気持ちで書いてました。
2. 「ディレクトリを配置して静的に公開（nodeのルーティングに含めない）してルートディレクトリを返す」の意味がよくわかりませんが、受け取ったファイルをどこかに配置します。エンドポイントは固定でいいと思うのでよく考えたら返す必要ないかもしれません。アップロードしたファイルはindex.htmlが基本的にルートにあると思うので、GETリクエストを受け付けたらそれを返すエンドポイントを作っとけばよいと思います。
3. あってます。実践的にはルートディレクトリのオブジェクトは以下のようになると思います。
`const source_code = {
        dirname: "/",
        file: [index],
        dir: [js, css, imgs]
    };`

**Reply by namikis** [reply]
回答ありがとうございます。
自分がフロントに詳しくないのもあるんですが、ReactやVueのコードが提出されたときにちゃんと動くのかなという不安がありました。
const source_code = { dirname: "/", file: [index], dir: [js, css, imgs] };
提出が必ずこの形になるなら問題はないですかね

**Reply by kenkenpa2126** [reply]
動くディレクトリ構造を作製するのも含めてテストなので、そこは気にしないで大丈夫だと思います。僕もフロントのことはよくわかりませんが、ReactやVueはnpmでインストールできるフレームワークなんで、フロントでターミナル動かして環境構築したうえでファイルアップロードしてもらうんで、フロントで動けば動くはずです。nodeのバージョンとかによって差異が生まれるかもしれませんが。
